From ff03ff7d31dc6ccddccad44abfa6e9e4b72c5b01 Mon Sep 17 00:00:00 2001
From: Nikita Tokarchuk <nikita@tokarch.uk>
Date: Wed, 6 Apr 2022 17:54:55 +0200
Subject: [PATCH] build: use mariadb libraries as a mysql connector

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index d71a4d7..64f94a2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -129,7 +129,7 @@ AC_ARG_ENABLE([postgres],
 if test "$enable_mysql" == yes
 then
 	# Check for mysql.
-	CFLAGS="$CFLAGS `pkg-config --cflags mysqlclient`"
+	CFLAGS="$CFLAGS `pkg-config --cflags libmariadb`"
 
 	# Check MySQL development header
 	AC_CHECK_HEADERS([mysql.h], , [AC_MSG_ERROR([[mysql header not found; install mysql-devel and dependencies for MySQL Support]])])
@@ -137,7 +137,7 @@ then
 	AC_DEFINE([HAVE_MYSQL], 1, [Define if mysql support is wanted])
 
 	# [] quotes, or autofoo will strip the character class [] in sed, breaking the regex.
-	[LIBMYSQLCLIENT_SO=$( readlink $(pkg-config --variable=libdir mysqlclient)/libmysqlclient.so | sed -e 's,^.*/,,;s/\(\.so\.[0-9]*\)\..*$/\1/')]
+	[LIBMYSQLCLIENT_SO=$( readlink $(pkg-config --variable=libdir libmariadb)/libmariadb.so | sed -e 's,^.*/,,;s/\(\.so\.[0-9]*\)\..*$/\1/')]
 	test -n "$LIBMYSQLCLIENT_SO" || AC_MSG_ERROR([Could not determine library name to be used in dlopen for mysql support])
 fi
 AC_DEFINE_UNQUOTED([LIBMYSQLCLIENT_SO], ["$LIBMYSQLCLIENT_SO"], [library name to be used in dlopen for mysql support])
-- 
2.35.1

